src/CoamClient.js

```diff
@@ -406,7 +406,7 @@ class CoamClient {
     }
 
     getUsersWithPermission(resourceType, resourceIdentifier, permission) {
-        let url = this.__buildUrl(`/auth/access-management/v1/search/canonicalPrincipals/byPermission?resource_type={{resourceType}}&resource_identifier={{resourceIdentifier}}&permission={{permission}}`, {
+        let url = this.__buildUrl(`/auth/access-management/v1/search/canonicalPrincipals/byPermission?resource_type={{resourceType}}${resourceIdentifier ? '&resource_identifier={{resourceIdentifier}}' : ''}&permission={{permission}}`, {
```

Comment:
Is this possible because `resourceIdentifier` is an optional parameter on the API? If so, why not having an `if` condition that simply doesn't add this query parameter at all in such case?

I guess the handling of empty strings is a really sensitive topic depending on a language so I don't immediately know what's going to happen when I read this code...

Reply:
Ok I was not reading it right last night, sorry.